src/libfetchers/git.cc

```diff
@@ -148,8 +148,14 @@ std::vector<PublicKey> getPublicKeys(const Attrs & attrs)
     std::vector<PublicKey> publicKeys;
     if (attrs.contains("publicKeys")) {
         nlohmann::json publicKeysJson = nlohmann::json::parse(getStrAttr(attrs, "publicKeys"));
-        ensureType(publicKeysJson, nlohmann::json::value_t::array);
-        publicKeys = publicKeysJson.get<std::vector<PublicKey>>();
+        auto pubKeys = getArray(publicKeysJson);
+        publicKeys.clear();
+        for (auto jsonKey : pubKeys) {
+            auto keyObj = getObject(jsonKey);
+            auto type = getString(getNullable(keyObj, "type").value_or("ssh-ed25519"));
+            auto key = getString(valueAt(keyObj, "key"));
+            publicKeys.push_back({ type, key });
+        }
```

Comment:
We shouldn't inline things like this. Instead we can change the JSON deserializer for `PublicKey` to use the new combinators. (And we could test that deserializer too.)

Reply:
Just that I understand that correctly this means implementing a new static method for `PublicKey::fromJSON`? ^^'